lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -197,12 +198,13 @@ WasmBytecodeGenerator::GenerateFunction()
 
         WasmOp op = wnLIMIT, newOp;
         EmitInfo exprInfo;
+        EnterEvalStackScope();
         while ((newOp = m_reader->ReadExpr()) != wnFUNC_END)
         {
-            ReleaseLocation(&exprInfo);
             op = newOp;
             exprInfo = EmitExpr(op);
         }
+        ExitEvalStackScope();
```

Comment:
You are exiting the eval scope too soon. We might emit a return before leaving that function.